server/src/sync.ts

```diff
@@ -38,7 +38,16 @@ export async function initSync(id: string, syncOptions: SyncOptions) {
       continue;
 
     for (const phoneNumber of googleContact.numbers) {
-      const whatsappContactId = whatsappContacts.get(phoneNumber);
+      let whatsappContactId = whatsappContacts.get(phoneNumber);
+
+      // Fix for Brazilian numbers with extra '9'
+      if (!whatsappContactId && phoneNumber.slice(0, 2) === '55') {
+        if (phoneNumber.length === 12) {
+          whatsappContactId = whatsappContacts.get(phoneNumber.slice(0, 4) + '9' + phoneNumber.slice(4));
+        } else {
+          whatsappContactId = whatsappContacts.get(phoneNumber.slice(0, 4) + phoneNumber.slice(5));
```

Comment:
Why do you remove a digit if the length is not 12?

Reply:
Previously, cell phone numbers had 12 digits total: 55 (XX) YYYY-YYYY. But now they have 13: 55 (XX) 9YYYY-YYYY. The idea is, if there was no match for the phone number and it had 12 digits, it might be missing a '9'. Otherwise, the phone number on Google already might have the extra '9', but the number on WhatsApp is missing it, so it is necessary to remove the extra number.